### DISCUSSION TITLE
Integrate the notion of battery - feedback needed

### DISCUSSION THREAD

**Discussion by jmcollin78** [jmcollin78/solar_optimizer#45] [Ideas]
I think Solar Optimizer should integrate the battery in case of physical installation with a battery.

What I propose it to add a feature to be able to prioritize to battery charge versus starting devices.

This could achieve by specifying the entityId of the battery soc (state of charge in %) and a minimal threshold of soc. If minimal threshold is not reach, Solar Optimizer will not start the device even if the solar production is sufficient, in order to let the battery charge.

Do you have any comment or suggestions regarding this new feature proposal ?

**Comment by sjordan0228** [comment]
Hi jmcollin78 i would be happy to test this out for you and give you some feedback. I have been searching high and low for automations around solar and battery setups without the ability to export. My setup is as follow:

EG4 18kpv inverter- Hybrid inverter that can export. However I do not want to ever do this
18.5KW of Solar panels
4 x  14.3 KWh batteries (total of 57.2KWh of storage)
Ford Lightning Truck and a Emporia 48a charger

I have a perfect use case. I have already sort of hacked together something that works but it would be much nicer to have your piece do this instead of my hack.

Here is my use case: When the truck is hooked up to the charger I would like for it to start charging when the battery SOC is at 100%. Start the charging off at 6 amps and monitor the battery power. Raise the charging amps by 4 (or whatever) until the battery power is in the negative (meaning its discharging). The idea is to find the sweet spot of sending the excess solar into the truck instead of pulling from the battery. I would like for it to check maybe every minute and to dynamically go up or down. If it hits a state where it cannot go up and has gone all the way down to the minimum (6 amps) then let it run until the battery soc is at 90%. 

Would also be nice to control my thermostats based off of excess or lack of excess. Before I got the truck I used to manually set our ac's in the summer to cool the house down more then normal to take advantage of the excess

Anyway if you are interested let me know.

**Reply by jmcollin78** [reply]
Hello, The feature is online now. You can test it and see if its solve your use case.
